subsys/bluetooth/host/adv.c

```diff
@@ -269,6 +269,15 @@ void bt_le_ext_adv_foreach(void (*func)(struct bt_le_ext_adv *adv, void *data),
 #endif /* defined(CONFIG_BT_EXT_ADV) */
 }
 
+void bt_adv_reset_adv_pool(void)
+{
+#if defined(CONFIG_BT_EXT_ADV)
+	(void)memset(&adv_pool, 0, sizeof(adv_pool));
+#else
+	(void)memset(&bt_dev.adv, 0, sizeof(bt_dev.adv));
+#endif /* defined(CONFIG_BT_EXT_ADV) */
```

Comment:
I think we want to also zero out `bt_dev.adv` when `CONFIG_BT_EXT_ADV=y`. I suspect `adv_get_legacy` will malfunction otherwise.

```suggestion
#endif /* defined(CONFIG_BT_EXT_ADV) */
	(void)memset(&bt_dev.adv, 0, sizeof(bt_dev.adv));
```